packages/node-core/src/indexer/fetch.service.ts

```diff
@@ -278,25 +258,65 @@ export abstract class BaseFetchService<DS extends BaseDataSource, B extends IBlo
     }
   }
 
+  // get all modulo numbers with a specific block ranges
+  private getModuloBlocks(startHeight: number, endHeight: number): number[] {
+    // no modulos in the filters been found
+    if (this.getModulos(this.projectService.getAllDataSources()).length < 0) return [];
+    // Find relevant ds
+    const {endHeight: rangeEndHeight, value: relevantDS} = this.getRelevantDsDetails(startHeight);
+    // Min of current ds endHeight
+    const minDsEndHeight = Math.min(...relevantDS.map((d) => d?.endBlock ?? Number.MAX_SAFE_INTEGER));
```

Comment:
I think this is redundant, doesn't `rangeEndHeight` always return the same or less value?

Reply:
Yes, looks like a mistake here. I thought `rangeEndHeight` would be next dataSource startHeight -1.